src/citrine/_rest/collection.py

```diff
@@ -79,11 +80,56 @@ def list(self,
             Resources in this collection.
 
         """
-        return self._paginator.paginate(self._fetch_page,
+        return self._paginator.paginate(self._fetch_page_list,
                                         self._build_collection_elements,
                                         page,
                                         per_page)
 
+
+    def search(self, search_params: Optional[dict] = None, 
+                    page: Optional[int] = None, per_page: int = 100) -> Iterable[ResourceType]:
+        """
+            Search for elements matching search_params in the collection, and paginate over
+            the results. This differs from the list function, because it makes a POST request
```

Comment:
In order to be able to search by different features by exact or substring match, it was easier to implement this as a POST request rather than a GET request with many query parameters